File: src/main/java/org/example/expert/domain/user/repository/UserRepository.java
package org.example.expert.domain.user.repository;

import org.example.expert.domain.user.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByEmail(String email);
    boolean existsByEmail(String email);

    @Query("SELECT u FROM User u WHERE u.userName = :nickname")
    Page<User> findByUserNameExact(@Param("nickname") String nickname, Pageable pageable);
}

File: src/main/java/org/example/expert/domain/user/repository/UserRepository.kt
package org.example.expert.domain.user.repository

import org.example.expert.domain.user.entity.User
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

import java.util.Optional

interface UserRepository : JpaRepository<User, Long> {

    fun findByEmail(email: String): Optional<User>

    fun existsByEmail(email: String): Boolean

    @Query("SELECT u FROM User u WHERE u.userName = :nickname")
    fun findByUserNameExact(@Param("nickname") nickname: String, pageable: Pageable): Page<User>
}
